src/AutoEnrollPlugin/Service/ValidationService.cs

```diff
@@ -0,0 +1,243 @@
+                (input.streamMapping.Sharpness?.Max ?? Double.MaxValue) >= input.notification.Sharpness)
+            {
+                return true;
+            }
+
+            return false;
+        };
+
+        private static readonly Func<(Notification notification, StreamMapping streamMapping), bool> validateYawAngle = (input) =>
+        {
+            if (input.notification.YawAngle == null)
+            {
+                return true;
+            }
+
+            if (
+                (input.streamMapping.YawAngle?.Min ?? Double.MinValue) <= input.notification.YawAngle &&
+                (input.streamMapping.YawAngle?.Max ?? Double.MaxValue) >= input.notification.YawAngle)
+            {
+                return true;
+            }
+
+            return false;
+        };
+
+
+        private static readonly Func<(Notification notification, StreamMapping streamMapping), bool> validateRollAngle = (input) =>
+        {
+            if (input.notification.RollAngle == null)
+            {
+                return true;
+            }
+
+            if (
+                (input.streamMapping.RollAngle?.Min ?? Double.MinValue) <= input.notification.RollAngle &&
+                (input.streamMapping.RollAngle?.Max ?? Double.MaxValue) >= input.notification.RollAngle)
+            {
+                return true;
+            }
+
+            return false;
+        };
+
+
+        private static readonly Func<(Notification notification, StreamMapping streamMapping), bool> validatePitchAngle = (input) =>
+        {
+            if (input.notification.PitchAngle == null)
+            {
+                return true;
+            }
+
+            if (
+                (input.streamMapping.PitchAngle?.Min ?? Double.MinValue) <= input.notification.PitchAngle &&
+                (input.streamMapping.PitchAngle?.Max ?? Double.MaxValue) >= input.notification.PitchAngle)
+            {
+                return true;
+            }
+
+            return false;
+        };
+
+        private static readonly Func<(Notification notification, StreamMapping streamMapping), bool>[] validateAll = new[] {
+            validateFaceQuality,
+            validateTemplateQuality,
+            validateFaceSize,
+            validateFaceArea,
+            validateFaceOrder,
+            validateFacesOnFrameCount,
+            validateBrightness,
+            validateSharpness,
+            validateYawAngle,
+            validateRollAngle,
+            validatePitchAngle
+        };
+
+        public ValidationService(
+            ILogger logger
+        )
+        {
+            this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        public bool Validate(Notification notification, StreamMapping streamMapping)
+        {
+            this.logger.Information("Face attributes: faceQuality {faceQuality}, templateQuality {templatequality}, faceSize {faceSize}, yawAngle {yawAngle}, rollAngle {rollAngle} pitchAngle {pitchAngle}", 
```

Comment:
nit: ignore for now but for future template log property names should be UpperCase also.